https://github.com/popcornmix/omxplayer/issues/297

Not a proper fix, but works as an ugly hack to workaround the issue.

--- omxplayer-a4ee0743ecf49ffa1818510e8cc7872282ca66a8/omxplayer.cpp.orig
+++ omxplayer-a4ee0743ecf49ffa1818510e8cc7872282ca66a8/omxplayer.cpp
@@ -1683,6 +1683,10 @@
 
     if(m_omx_reader.IsEof() && !m_omx_pkt)
     {
+      // EOF before playback started, stop.
+      if (m_av_clock->OMXIsPaused())
+        break;
+
       // demuxer EOF, but may have not played out data yet
       if ( (m_has_video && m_player_video.GetCached()) ||
            (m_has_audio && m_player_audio.GetCached()) )
